packages/console/src/pages/TenantSettings/TenantBasicSettings/index.tsx

```diff
@@ -1,54 +1,57 @@
-import { type TenantInfo, TenantTag } from '@logto/schemas/models';
+import { type TenantInfo, type TenantTag } from '@logto/schemas/models';
 import classNames from 'classnames';
-import { useEffect, useState } from 'react';
+import { useContext, useEffect, useMemo, useState } from 'react';
 import { FormProvider, useForm } from 'react-hook-form';
 import { toast } from 'react-hot-toast';
 import { useTranslation } from 'react-i18next';
 
 import { useCloudApi } from '@/cloud/hooks/use-cloud-api';
+import { useCloudSwr } from '@/cloud/hooks/use-cloud-swr';
 import AppError from '@/components/AppError';
 import AppLoading from '@/components/AppLoading';
 import PageMeta from '@/components/PageMeta';
 import SubmitFormChangesActionBar from '@/components/SubmitFormChangesActionBar';
 import UnsavedChangesAlertModal from '@/components/UnsavedChangesAlertModal';
-import useTenants from '@/hooks/use-tenants';
+import { TenantsContext } from '@/contexts/TenantsProvider';
 
 import DeleteCard from './DeleteCard';
 import DeleteModal from './DeleteModal';
 import ProfileForm from './ProfileForm';
 import * as styles from './index.module.scss';
 import { type TenantSettingsForm } from './types.js';
 
-const tenantProfileToForm = (tenant?: TenantInfo): TenantSettingsForm => {
+const tenantProfileToForm = (tenant: TenantInfo): TenantSettingsForm => {
+  const { name, tag } = tenant;
   return {
-    profile: { name: tenant?.name ?? 'My project', tag: tenant?.tag ?? TenantTag.Development },
+    profile: { name, tag },
   };
 };
 
 function TenantBasicSettings() {
   const { t } = useTranslation(undefined, { keyPrefix: 'admin_console' });
   const api = useCloudApi();
-  const {
-    currentTenant,
-    currentTenantId,
-    error: requestError,
-    mutate,
-    isLoading,
-    tenants,
-  } = useTenants();
+  const { data: availableTenants, error: requestError, mutate } = useCloudSwr('/api/tenants');
+  const { tenants, setTenants, currentTenantId, navigate } = useContext(TenantsContext);
```

Comment:
I think we can totally remove `TenantsContext`. Just load tenants from swr, and get current tenant ID from URL

Reply:
We use `currentTenantId` (which is stored in TenantsContext) to control the switching between different tenants. Some centralized tenant fallback logics are in TenantsContext. We do not want to handle the fallback logic everywhere.
Also, tenants context is a top-level context, which determines whether the current state is prepared to enter cloud app.